src/main/java/seedu/address/model/transaction/Transaction.java

```diff
@@ -64,21 +86,27 @@ public boolean equals(Object other) {
         }
 
         Transaction otherTransaction = (Transaction) other;
-        return otherTransaction.getPayer().equals(this.getPayer())
+        return otherTransaction.getId().equals(this.getId())
```

Comment:
By definition, getId().equals(this.getId()) should be enough for the method equals ?

Reply:
Thanks, I'll change that